source/imex_stepper.cc

```diff
@@ -156,9 +159,28 @@ unsigned int IMEXStepper<VEC>::start_ode(VEC &solution)
       unsigned int outer_iter = 0;
       unsigned int nonlin_iter = 0;
       interface.residual(t, solution, *solution_dot, *residual);
-      double res_norm = residual->l2_norm();
+      double res_norm = 0.0;
+      double solution_norm = 0.0;
+
+      if (norm=="l2")
+        {
+          if (abs_tol>0.0||rel_tol>0.0)
+            res_norm = residual->l2_norm();
+          if (rel_tol>0.0)
+            solution_norm = solution.l2_norm();
+        }
+      else if (norm=="linfty")
+        {
+          if (abs_tol>0.0||rel_tol>0.0)
+            res_norm = residual->linfty_norm();
+          if (rel_tol>0.0)
+            solution_norm = solution.linfty_norm();
+        }
+
       // The nonlinear solver iteration cycle begins here.
-      while (outer_iter < max_outer_non_linear_iterations && res_norm > abs_tol)
+      while (outer_iter < max_outer_non_linear_iterations &&
+             res_norm > abs_tol &&
+             res_norm > rel_tol*solution_norm)
```

Comment:
Certo! È sempre vera a meno che non hai errore nullo, ma in quel caso it is a Dream... No?

--
Mauro

> Il giorno 14 dic 2015, alle ore 10:15, Alberto Sartori <notifications@github.com> ha scritto:
> 
> In source/imex_stepper.cc:
> 
> >        // The nonlinear solver iteration cycle begins here.
> > -      while (outer_iter < max_outer_non_linear_iterations && res_norm > abs_tol)
> > +      while (outer_iter < max_outer_non_linear_iterations &&
> > +             res_norm > abs_tol &&
> > +             res_norm > rel_tol*solution_norm)
> sei sicuro che entri nel ciclo se lasci rel_tol=0?
> 
> —
> Reply to this email directly or view it on GitHub.
>